superset-frontend/src/components/Select/styles.tsx

```diff
@@ -272,6 +297,24 @@ export const DEFAULT_COMPONENTS: SelectComponentsConfig<any> = {
       />
     </DropdownIndicator>
   ),
+  Input: (props: InputProps) => {
+    const {
+      isMulti,
+      value,
+      placeholder,
+    }: {
+      isMulti?: boolean;
+      value?: ValueType<any>;
+      placeholder?: ReactNode;
```

Comment:
I think both `isMulti` and `value` are actually [already available](https://github.com/JedWatson/react-select/blob/d7c866d1f212223d4731e83f61f4d354fc6810a6/packages/react-select/src/Select.js#L1451-L1472) to `InputProps`, just not added to `@types/react-select`'s official [InputProps](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-select/src/components/Input.d.ts#L7-L16). You should get a clean typing if you just add them to your custom `InputProps` above.

```typescript
type InputProps = ReactSelectInputProps & {
  isMulti?: boolean;
  value?: ValueType<any>
  selectProps: SelectProps<any>;
};

// ....

 Input: ({ isMulti, value, selectProps: { placeholder } }: InputProps) => {
     // ...
 },
```

The reason why you could't directly use `selectProps` below is because you've set it as an optional prop (sorry for my misleading example!). Changing

```
  selectProps?: SelectProps<any>;
```

to 

```
  selectProps: SelectProps<any>;
```

would do the trick.

Reply:
This did.. thanks! I just had one more issue with value.length that still needed to be asserted, but much better.